sources/TileDB.CSharp/FragmentInfo.cs

```diff
@@ -0,0 +1,681 @@
+        /// <param name="minimumBoundedRectangleIndex">The index of the MBR of interest.</param>
+        /// <param name="dimensionName">The name of the dimension of interest.</param>
+        /// <returns>The start and end values of the MBR, inclusive.</returns>
+        /// <exception cref="NotSupportedException"><typeparamref name="T"/> is unsupported.</exception>
+        /// <exception cref="InvalidOperationException"><typeparamref name="T"/> is incompatible
+        /// with the dimension's data type.</exception>
+        public (T Start, T End) GetMinimumBoundedRectangle<T>(uint fragmentIndex, uint minimumBoundedRectangleIndex, string dimensionName)
+        {
+            DataType dataType = GetDimensionType(fragmentIndex, dimensionName);
+            ValidateDomainType<T>(dataType);
+            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+            {
+                if (typeof(T) == typeof(string))
+                {
+                    (string startStr, string endStr) =
+                        GetStringMinimumBoundedRectangle(fragmentIndex, minimumBoundedRectangleIndex, dimensionName, dataType);
+                    return ((T)(object)startStr, (T)(object)endStr);
+                }
+                ThrowHelpers.ThrowTypeNotSupported();
+                return default;
+            }
+
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            using var ms_dimensionName = new MarshaledString(dimensionName);
+            // We always allocate twice the maximum possible data type to avoid buffer overflows.
+            // If the dimension is of type uint64 and we pass a buffer of two bytes, it will write past the buffer's boundaries.
+            byte* mbr = stackalloc byte[sizeof(ulong) * 2];
+            _ctx.handle_error(Methods.tiledb_fragment_info_get_mbr_from_name(ctxHandle,
+                handle, fragmentIndex, minimumBoundedRectangleIndex, ms_dimensionName, mbr));
+
+            var start = Unsafe.ReadUnaligned<T>(mbr);
+            var end = Unsafe.ReadUnaligned<T>(mbr + Unsafe.SizeOf<T>());
+            return (start, end);
+        }
+
+        private (string Start, string End) GetStringMinimumBoundedRectangle(uint fragmentIndex, uint minimumBoundedRectangleIndex, uint dimensionIndex, DataType dataType)
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            ulong startSize64, endSize64;
+            _ctx.handle_error(Methods.tiledb_fragment_info_get_mbr_var_size_from_index(ctxHandle,
+                handle, fragmentIndex, minimumBoundedRectangleIndex, dimensionIndex, &startSize64, &endSize64));
+            int startSize = checked((int)startSize64);
+            int endSize = checked((int)endSize64);
+            using var startBuf = new ScratchBuffer<byte>(startSize, stackalloc byte[128]);
```

Comment:
Does this implicitly mean that the maximum size is 128? Or will ScratchBuffer spill to the heap if larger?

Reply:
Ok, I see how it works - nice 💯